core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/AbstractFileBatchSource.java

```diff
@@ -181,7 +186,22 @@ public void prepareRun(BatchSourceContext context) throws Exception {
       context.setInput(Input.of(config.referenceName, new SourceInputFormatProvider(config.inputFormatClass, conf)));
     }
 
+    List<Schema.Field> fields = null;
 
+    if (config.getSchema() != null) {
+       fields = config.getSchema().getFields();
+    } else if ("text".equalsIgnoreCase(config.format)) {
+      fields = PathTrackingInputFormat.getTextOutputSchema(config.pathField).getFields();
+    }
+
+    if (fields != null) {
+      FieldOperation operation = new FieldReadOperation("Read", "Reading files.",
+                                                        EndPoint.of(context.getNamespace(), config.referenceName),
+                                                        fields.stream().map(Schema.Field::getName)
+                                                          .collect(Collectors.toList())
+      );
```

Comment:
put this on the previous line to be consistent

Reply:
fixed.